src/server/native_controllers/native_controller.js

```diff
@@ -0,0 +1,41 @@
+import { Observable, ReplaySubject } from 'rxjs';
+
+export class NativeController {
+
+  killed = false;
```

Comment:
In my experience, properties like this change over time and without the help of something like Typescript it's really easy to miss a place that was looking for `nativeController.killed` which is now always going to be `undefined`. This is why I lean pretty heavily on getters like `nativeController.isKilled()`, which will be much more noisy if it gets renamed down the road and we miss a use somewhere.